### PR TITLE
customize search engine

### PR DESCRIPTION
**Pull request by Nriver**
related issue #3882

### CODE REVIEW

File: src/public/app/widgets/type_widgets/options/other/search_engine.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+</div>`;
+
+const SEARCH_ENGINES = {
+    "Bing": "https://www.bing.com/search?q={keyword}",
+    "Baidu": "https://www.baidu.com/s?wd={keyword}",
+    "Duckduckgo": "https://duckduckgo.com/?q={keyword}",
+    "Google": "https://www.google.com/search?q={keyword}",
+}
+
+export default class SearchEngineOptions extends OptionsWidget {
```
Comment (soulsands) [inline src/public/app/widgets/type_widgets/options/other/search_engine.js:46]: this feature only works in desktop build. Runtime should be checked here.
```
 isEnabled() {
        return super.isEnabled() && utils.isElectron();
    }
```

Comment (Nriver) [reply src/public/app/widgets/type_widgets/options/other/search_engine.js:46]: You are right. The check has been added.

**Comment by soulsands** [comment]
These two options can be add to defaultOptions, similar to this commit    [  src\services\options_init.js  ](https://github.com/zadam/trilium/commit/6f238f5f1aec7dac83ca1e69441c66e06e220928)

**Comment by Nriver** [comment]
> These two options can be add to defaultOptions, similar to this commit [ src\services\options_init.js ](https://github.com/zadam/trilium/commit/6f238f5f1aec7dac83ca1e69441c66e06e220928)

Thanks! This prevents the bug with undefined values.

**Comment by zadam** [comment]
Thanks!

**Pull request merged**